core/domain/exp_services.py

```diff
@@ -1926,9 +1941,24 @@ def are_changes_mergeable(exp_id, frontend_version, change_list):
         }
 
         if len(added_state_names) > 0 or len(deleted_state_names) > 0:
-            # Here we will send the changelist, version, latest_version,
-            # and exploration to the admin, so that the conditions
-            # can be reviewed.
+            # In case of the addition and the deletion of the state,
+            # we are rejecting the mergebility because these cases
+            # change the flow of the exploration and are quite complex
+            # for now to handle. So in such cases, we are sending the
+            # changelist, frontend_version, backend_version and
+            # exploration id to the admin, so that we can look into the
+            # situations and can figure out the way if it’s possible to
+            # handle these cases.
+            if feconf.CAN_SEND_EMAILS:
+                email_manager.send_mail_to_admin(
+                    'Adding/Deleting State Changes not mergeable',
+                    'The exploration id is: %s, \n '
+                    'The frontend version is: %s \n '
+                    'The backend version is: %s \n'
+                    'The changes list is: %s '
+                    % (exp_id, frontend_version,
+                       backend_version_exploration.version,
+                       change_list))
```

Comment:
For the part about "wording", I would suggest something along these lines:

```
Some draft changes were rejected in exploration xxx because state(s) were added/deleted to the exploration before the draft could be saved. This resulted in a conflict. Please see the rejected change list below:

Discarded changelist: zzz

Frontend version: xxx
Bakend version: yyy
```

Reply:
Made the changes.